### PR TITLE
Enable configurator read-only view for quote display

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts [hunk 1]
```diff
@@ -31,14 +31,18 @@ export class ConfigureCartEntryComponent {
...
-    return this.cartEntry.orderCode !== undefined
-      ? CommonConfigurator.OwnerType.ORDER_ENTRY
-      : CommonConfigurator.OwnerType.CART_ENTRY;
+    if (this.cartEntry.quoteCode !== undefined) {
+      return CommonConfigurator.OwnerType.QUOTE_ENTRY;
+    } else {
+      return this.cartEntry.orderCode !== undefined
+        ? CommonConfigurator.OwnerType.ORDER_ENTRY
+        : CommonConfigurator.OwnerType.CART_ENTRY;
+    }
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:45]: so if there is a quoteCode and orderCode, we still assume QUOTE_ENTRY. Is this desired and if yes a test for this case would be nice

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts [hunk 2]
```diff
@@ -53,14 +57,21 @@ export class ConfigureCartEntryComponent {
...
       : entryNumber.toString();
   }
 
+  protected getCode(): string | undefined {
+    if (this.cartEntry.quoteCode) {
+      return this.cartEntry.quoteCode;
+    } else if (this.cartEntry.orderCode) {
+      return this.cartEntry.orderCode;
+    }
+    return undefined;
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts]: same as for OWNER_TYPE: is it desired that we return the quoteCode if both  quote  and order code are defined? If yes a test for this case would be good.

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts (hunk 1 above)
Comment (Uli-Tiger) [inline feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts]: so if there is a quoteCode and orderCode, we still assume QUOTE_ENTRY. Is this desired and if yes a test for this case would be nice

File: feature-libs/quote/tsconfig.lib.json
Comment (Uli-Tiger) [inline feature-libs/quote/tsconfig.lib.json:1]: changes are intended?

**Review by Uli-Tiger — CHANGES_REQUESTED** [review]
some remarks based on a first look - need to check in more detail tomorrow

**Comment by Uli-Tiger** [comment]
@ChristophHi 
  FAIL  src/shared/utils/graph-utils_spec.ts
  ● Graph utils › library dependency graph › should have generated the correct order

File: feature-libs/quote/tsconfig.lib.json
Comment (ChristophHi) [reply feature-libs/quote/tsconfig.lib.json:1]: @Uli-Tiger Yes those changes are intended. Somehow a dependency to product-configurator was introduced before (done by the teams that dealt with that before) which does not make sense. Removing it clears a number of other dependencies as well)

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts (hunk 2 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts]: Agree will improve that

**Comment by ChristophHi** [comment]
> @ChristophHi FAIL src/shared/utils/graph-utils_spec.ts ● Graph utils › library dependency graph › should have generated the correct order

Probably also because I resolved the dependencies between quote and product-configurator

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts [hunk 3]
```diff
@@ -31,14 +33,24 @@ export class ConfigureCartEntryComponent {
...
    * @returns - an owner type
    */
   getOwnerType(): CommonConfigurator.OwnerType {
-    return this.cartEntry.orderCode !== undefined
-      ? CommonConfigurator.OwnerType.ORDER_ENTRY
-      : CommonConfigurator.OwnerType.CART_ENTRY;
+    if (this.cartEntry.quoteCode || this.cartEntry.orderCode) {
+      if (!this.cartEntry.quoteCode) {
+        return CommonConfigurator.OwnerType.ORDER_ENTRY;
+      }
```
Comment (steinsebastian) [inline feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:44]: Why is the if-statement negative?
Why not using:
      if (this.cartEntry.quoteCode) {
        return CommonConfigurator.OwnerType.QUOTE_ENTRY;
      }

**Review by Uli-Tiger — APPROVED** [review]

File: feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts (hunk 3 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/common/components/configure-cart-entry/configure-cart-entry.component.ts:44]: @steinsebastian Please see the line 41 above, and also consider that we throw an error if both codes are provided (which is an illegal state)

**Pull request merged**